### ISSUE TITLE
macOS Catalina: Cannot connect to the Extensions Marketplace (Error querying extensions gallery connect ENOENT)

### ISSUE THREAD

**Issue by Sebi94nbg** [microsoft/vscode#85402]
- VSCode Version: 1.40.1 (also tested with 1.41.0-insider)
- OS Version: (latest) macOS Catalina 10.15.1 (19B88) on a MacBook Pro 13" (Early 2015)

I couldn't find any reported issue, which helped me to fix my issue.

Tested stable version:
```
Version: 1.40.1
Commit: 8795a9889db74563ddd43eb0a897a2384129a619
Date: 2019-11-13T16:47:44.719Z
Electron: 6.1.2
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Darwin x64 19.0.0
```

Tested insider version:
```
Version: 1.41.0-insider
Commit: 599c076d91be1374cf51004cec610f3bcaf4c9cd
Date: 2019-11-22T07:19:06.796Z
Electron: 6.1.5
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Darwin x64 19.0.0
```

I've simply downloaded the latest version of Visual Studio Code, allowed access to all directories and disallowed access to my microphone.

After starting it, I wanted to install my needed extensions, but the search always says the same:

> We cannot connect to the Extensions Marketplace at this time, please try again later.

I have this issue since starting this week already. Only standard firewalls (FRITZ!Box router + macOS firewall) are between internet and VSCode.

The developer console of VSCode shows this:

![image](https://user-images.githubusercontent.com/5154682/69448489-ecd6ba80-0d58-11ea-8b6e-a8465e81cc5a.png)

> Error: connect ENOENT /Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support/Code/1.40.1-shared.sock
>     at PipeConnectWrap.afterConnect [as oncomplete] (net.js:1054)
> telemetryIpc.ts:35

> Error querying extensions gallery connect ENOENT /Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support/Code/1.40.1-shared.sock
(anonymous) @ extensionsViews.ts:254

>   ERR connect ENOENT /Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support/Code/1.40.1-shared.sock: Error: connect ENOENT /Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support/Code/1.40.1-shared.sock
>     at PipeConnectWrap.afterConnect [as oncomplete] (net.js:1054:14)
> log.ts:196

The here mentioned directory looks like this:

```
$ ls -lhart /Users/sebastiankraetzig/.composer/vendor/bin/Library/Application\ Support/Code/
total 24
-rw-r--r--  1 sebastiankraetzig  staff    36B 19 Nov 20:13 machineid
drwxr-xr-x  4 sebastiankraetzig  staff   128B 19 Nov 20:13 CachedExtensions
drwxr-xr-x  5 sebastiankraetzig  staff   160B 19 Nov 20:15 User
drwxr-xr-x  7 sebastiankraetzig  staff   224B 22 Nov 18:35 .
drwxr-xr-x  4 sebastiankraetzig  staff   128B 22 Nov 18:40 ..
-rw-r--r--  1 sebastiankraetzig  staff   447B 22 Nov 18:47 rapid_render.json
-rw-r--r--  1 sebastiankraetzig  staff     2B 22 Nov 18:47 languagepacks.json
```

I've also tested, if it's fixed, when I grant this application full disk access. Unfortunately not - same issue.

Does this issue occur when all extensions are disabled? Yes

I don't even have any extensions installed, since I can not search in the extensions marketplace.

**Comment by sandy081** [comment]
Looks like some of the VS Code processes are not running due to socket connection issues. 

@joaomoreno Is there an easy to way to get the user fix this other than restarting the machine?

**Comment by joaomoreno** [comment]
@Sebi94nbg  Please try restarting and seeing whether it fixes it. Also, what file system are you using?

**Comment by Sebi94nbg** [comment]
I've rebooted my MacBook, but the issue still persists.

I use the default Apple file system: APFS

FileVault is enabled, so it's "APFS (encrypted)".

**Comment by joaomoreno** [comment]
Oh I know! I remember this one! This feels like _obscure issue trivia_.

> Some operating systems, such as HP/UX, limit sun_path to only be up to 92 bytes. macOS, QNX and 4.4 BSD limit them to 104. Linux, 4.3 BSD and z/OS have a limit of 108 bytes. A mandatory terminating null byte is included in this calculation.

in https://kohlschutter.github.io/junixsocket/unixsockets.html

The problem here is that we're creating a Unix socket in the user data dir (necessary for a *lot* of things), which in your case is already pretty long:

```
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support/Code/
```

Can you maybe change it back to being the usual one?

```
/Users/sebastiankraetzig/Library/Application\ Support/Code/
```

**Comment by Sebi94nbg** [comment]
I didn't change or configure this path. It was the default path after an installation with the defaults. How can I change that? :D

**Comment by joaomoreno** [comment]
This is strange. Where did you acquire VS Code? What is this `composer`?

**Comment by Sebi94nbg** [comment]
I've downloaded the latest version from here: https://code.visualstudio.com/download (clicked on Mac download)

This composer thing had a reference to Laravel (framework), but in the vendor directory was only VSCode:

```
Sebastian-MBP:~ root# find /Users/sebastiankraetzig/.composer/vendor/bin/Library/Application\ Support/
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support/
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/CachedExtensions
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/CachedExtensions/user
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/CachedExtensions/builtin
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/rapid_render.json
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/snippets
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/settings.json
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/workspaceStorage
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/workspaceStorage/1574190822982
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/workspaceStorage/1574190822982/state.vscdb.backup
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/workspaceStorage/1574190822982/state.vscdb
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/workspaceStorage/f29c161852647a33053fc95b7ea4b1b5
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/workspaceStorage/f29c161852647a33053fc95b7ea4b1b5/state.vscdb.backup
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/workspaceStorage/f29c161852647a33053fc95b7ea4b1b5/workspace.json
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/workspaceStorage/f29c161852647a33053fc95b7ea4b1b5/state.vscdb
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/workspaceStorage/7ace6b4bf8eb65ad98e551599dba2064
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/workspaceStorage/7ace6b4bf8eb65ad98e551599dba2064/state.vscdb.backup
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/workspaceStorage/7ace6b4bf8eb65ad98e551599dba2064/workspace.json
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/User/workspaceStorage/7ace6b4bf8eb65ad98e551599dba2064/state.vscdb
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/languagepacks.json
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code/machineid
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code - Insiders
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code - Insiders/CachedExtensions
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code - Insiders/CachedExtensions/user
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code - Insiders/CachedExtensions/builtin
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code - Insiders/rapid_render.json
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code - Insiders/User
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code - Insiders/User/settings.json
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code - Insiders/User/workspaceStorage
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code - Insiders/User/workspaceStorage/1574444410474
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code - Insiders/User/workspaceStorage/1574444410474/state.vscdb.backup
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code - Insiders/User/workspaceStorage/1574444410474/state.vscdb
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code - Insiders/languagepacks.json
/Users/sebastiankraetzig/.composer/vendor/bin/Library/Application Support//Code - Insiders/machineid
```

I've now uninstalled the application like this:
# Open Finder
# Open "Applications" in Finder
# Mark "Visual Studio Code"
# Deleted (Uninstalled) the application using the key combination `cmd` and `delete / backspace`

After that I've also removed the entire `.composer` folder: `rm -rf /Users/sebastiankraetzig/.composer/`

And then I've rebooted once, downloaded again the latest version from https://code.visualstudio.com/download and installed it again:
# Unzipped the downloaded file `VSCode-darwin-stable.zip`
# Moved the resulted `Visual Studio Code` file to the `Applications` folder
# Opened this app the first time using `ctrl` -> right-click -> `Open`
# Confirmed macOS warning message regarding "can not verify" by clicking on `Open`

And I still have the same problem: I can't search for extensions.

![image](https://user-images.githubusercontent.com/5154682/69747355-841b8380-1146-11ea-8f3c-2d97f3114fc5.png)

And the before deleted `.composer` directory exists again:

```
$ ls -lh /Users/sebastiankraetzig/.composer/vendor/bin/Library/Application\ Support/Code/
total 24
drwxr-xr-x  4 sebastiankraetzig  staff   128B 27 Nov 18:46 CachedExtensions
drwxr-xr-x  3 sebastiankraetzig  staff    96B 27 Nov 18:45 User
-rw-r--r--  1 sebastiankraetzig  staff     2B 27 Nov 18:46 languagepacks.json
-rw-r--r--  1 sebastiankraetzig  staff    36B 27 Nov 18:46 machineid
-rw-r--r--  1 sebastiankraetzig  staff   447B 27 Nov 18:46 rapid_render.json
`
``

But this time without the `composer.json`:

```
$ find /Users/sebastiankraetzig/.composer/
/Users/sebastiankraetzig/.composer/
/Users/sebastiankraetzig/.composer//vendor
/Users/sebastiankraetzig/.composer//vendor/bin
/Users/sebastiankraetzig/.composer//vendor/bin/Library
/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support
/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support/Code
/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support/Code/CachedExtensions
/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support/Code/CachedExtensions/user
/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support/Code/CachedExtensions/builtin
/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support/Code/rapid_render.json
/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support/Code/User
/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support/Code/User/workspaceStorage
/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support/Code/User/workspaceStorage/f29c161852647a33053fc95b7ea4b1b5
/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support/Code/User/workspaceStorage/f29c161852647a33053fc95b7ea4b1b5/workspace.json
/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support/Code/User/workspaceStorage/f29c161852647a33053fc95b7ea4b1b5/state.vscdb
/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support/Code/languagepacks.json
/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support/Code/machineid
```

**Comment by joaomoreno** [comment]
Really not sure why that directory is created instead of the default. Do you have a `VSCODE_PORTABLE` variable set?

**Comment by Sebi94nbg** [comment]
No, I don't have this set. Is there a way to make sure, that I fully remove the application and all settings, caches and so on?

**Comment by joaomoreno** [comment]
Yes that's the way you just did it. But something else in your system is makign VS Code think that your Application Support directory is in `/Users/sebastiankraetzig/.composer//vendor/bin/Library/Application Support`... You have to go and find what that thing is.

Maybe from a terminal, run `env`, and see if you find any references to `.composer`.

**Comment by Sebi94nbg** [comment]
Mhmm, ok. I'll do it this evening.

**Comment by Sebi94nbg** [comment]
You were right. An environment variable was changed and included this `.composer` path:

```
$ env | grep -i composer
HOME=/Users/sebastiankraetzig/.composer/vendor/bin
```

It was configured in my `.bash_profile` file:

```
$ grep -i composer .bash_profile 
HOME=$HOME/.composer/vendor/bin
```

I've removed the entire line now. I'm not sure, why this was even there. Maybe I needed that years ago for something...

Anyway: After removing this line, I've rebooted my MacBook, opened VSCode and it worked! I am able to browse and search for extensions in the extensions marketplace! :)

![image](https://user-images.githubusercontent.com/5154682/69823908-9e1e9a00-120a-11ea-99cc-ee4271d404cb.png)

Thanks for your time and support!
